tap_twitter_ads/streams.py

```diff
@@ -308,6 +311,11 @@ def sync_endpoint(self,
             # API Call
             cursor = self.get_resource(stream_name, client, path, new_params)
 
+            # cursor is an object like a generator(yield). First, it will be iterated for the parent stream with 
+            # the parent's bookmark. But, for the child also we want to iterate through all parent records 
+            # based on the child bookmark and collect parent_ids. That's why we are making a cursor copy before the parent iteration.
+            cursor_child = copy.deepcopy(cursor) # Cursor for children to retrieve parent_ids
```

Comment:
We can't be able to make a copy of the generator so will the copy work for the cursor?

Reply:
Here, the cursor is not an exact generator. cursor is just an object that is available in the [twitter-sdk](https://github.com/twitterdev/twitter-python-ads-sdk/blob/master/twitter_ads/cursor.py). That's why we can make copy of cursor and we have verified that it is working totally fine.